src/chains/ethereum/ethereum/src/api-types.ts

```diff
@@ -0,0 +1,195 @@
+>[1]["data"];
+
+type AsCall<T> = Flatten<
+  Omit<T, "from"> & {
+    readonly from?: string;
+  }
+>;
+type AsPooled<T> = Flatten<
+  Omit<T, "blockNumber" | "blockHash" | "transactionIndex"> & {
+    blockNumber: null;
+    blockHash: null;
+    transactionIndex: null;
+  }
+>;
+
+type PublicPrivate = "public" | "private";
+
+/**
+ * Since our types come from all over the place and get smushed together and
+ * pulled apart, we "Flatten" (there is probably a
+ * better word) these type complexities by using a TypeScript trick:
+ * `Pick<T, keyof T>`. This picks all the keys (and their values) from T,
+ * resulting in the same type shape, but the intermediate types are all skipped
+ * and intersections are simplified.
+ *
+ * ```
+ * type SomeTypes = {prop: string, prop2: number};
+ * type Thing = Omit<SomeTypes, "prop2"> & {addProp: true};
+ * ```
+ * gets turned into
+ * ```
+ * type Thing = {prop: string, addProp: true}
+ * ```
+ */
+type Flatten<T> = Pick<T, keyof T>;
+
+/** Public types */
+export namespace Ethereum {
+  export type Provider = EthereumProvider;
+  export type Tag = keyof typeof InternalTag;
+
+  //#region trace/debug
+  export type TraceTransactionOptions = UtilTypes.TraceTransactionOptions;
+  export type TraceTransactionResult<P extends PublicPrivate = "public"> =
+    P extends "public"
+      ? Externalize<TraceTransactionResult<"private">>
+      : UtilTypes.TraceTransactionResult;
+  export type StorageRangeAtResult<P extends PublicPrivate = "public"> =
+    P extends "public"
+      ? Externalize<StorageRangeAtResult<"private">>
+      : UtilTypes.StorageRangeAtResult;
+  //#endregion
+
+  //#region subscriptions/filters
+  export type SubscriptionOptions = UtilTypes.BaseFilterArgs;
+  export type LogsFilter = UtilTypes.FilterArgs;
+  export type Filter = UtilTypes.RangeFilterArgs;
+  export type SubscriptionName = UtilTypes.SubscriptionName;
+  export type SubscriptionId = UtilTypes.SubscriptionId;
+  export type Logs = Log[];
+  //#endregion subscriptions/filters
+
+  //#region transactions
+  export namespace Transaction {
+    export type Legacy = Flatten<TransactionTypes.LegacyRpcTransaction>;
+    export type EIP1559 =
+      Flatten<TransactionTypes.EIP1559FeeMarketRpcTransaction>;
+    export type EIP2930 =
+      Flatten<TransactionTypes.EIP2930AccessListRpcTransaction>;
+
+    /**
+     * Transaction receipt returned from `eth_getTransactionReceipt`
+     */
+    export type Receipt<P extends PublicPrivate = "public"> = P extends "public"
+      ? Externalize<Transaction.Receipt<"private">>
+      : TransactionTypes.TransactionReceipt;
+  }
+
+  /**
+   * Transaction types sent to `eth_sendTransaction` and
+   * `personal_sendTransaction`
+   */
+  export type Transaction =
+    | Ethereum.Transaction.Legacy
+    | Ethereum.Transaction.EIP1559
+    | Ethereum.Transaction.EIP2930;
+  //#endregion transactions
+
+  //#region call/estimate
+  export namespace Call {
+    export type Transaction =
```

Comment:
It seems backwards and a little harder to follow to export the type before the namespace. Can we switch the order? (Just a slightly easier read, not very important)